andela/travel_tool_back
# #161656449 Fix room selection issue

#### What does this PR do?
- Fixes the issue with room selection based on gender issue

#### Description of Task to be completed?
- ensure that requesters cannot book for rooms occupied by the opposite gender
- ensure that beds not existing in the selected destination guesthouse cannot be booked

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Checkout the branch `git checkout bg-fix-bed-selection-161656449`
- Install dependencies using `yarn install`
- Rollback and run database migrations with `yarn db:rollmigrate`
- Start the server with `make start`
- Setup the frontend app with this - `https://github.com/andela/travel_tool_front`
- Visit https://travela-local.andela.com
- Login to the application
- Create a request setting your gender as `Male`
- Take note of the bed and room you selected
- Create another request with the same date range as the previous but setting your gender as `Female`
- The list of available rooms should not consist of any bed in the room which you selected in the previous request.

#### Any background context you want to provide?
- You should have available rooms in a guesthouse in order to test this bug fix

#### What are the relevant pivotal tracker stories?
[#161656449](https://www.pivotaltracker.com/story/show/161656449)

#### Screenshots (if appropriate)

#### Questions:
